apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.TargetedSessionPool.cs

```diff
@@ -242,10 +250,20 @@ private Task<PooledSession> GetSessionAcquisitionTask(ModeOneofCase transactionM
                     else
                     {
                         // Otherwise, drop out of the lock to create a session from scratch "inline" (still asynchronously, but not via an in-flight request).
+                        // This means basically to try and nurse the pool back to health.
                     }
                 }
                 // Effectively the "else" block above, but outside the lock
-                return CreatePooledSessionAsync(cancellationToken);
+
+                // Just to hide the "await" here so we don't have to make this method async.
+                // Because we cannot await inside the lock, the only way to make this method
+                // async is to have it return Task<Task<PooledSession>> which is weird.
+                return Task.Run(async () =>
+                {
+                    await GetNursePoolBackToHealthTask(cancellationToken).ConfigureAwait(false);
+                    // If we reached this point the call above dind't fail and the pool is healthy.
```

Comment:
Nit: dind't => didn't (or make it "The nursing task succeeded and the pool is healthy"

Reply:
Also, it's possible that the above succeeds because the pool is shutdown. Is that still okay?